lightning-c-bindings/README.md

```diff
@@ -0,0 +1,143 @@
+
+ * Rust "unitary" enums are mapped simply as an equivalent C enum, however some Rust enums have
+   variants which contain payloads. Such enums are mapped automatically by cbindgen as a tag which
+   indicates the type and a union which holds the relevant fields for a given tag. A `X_free`
+   function is provided for the enum as a whole which automatically frees the correct fields for a
+   given tag, and a `Sentinel` tag is provided which causes the free function to do nothing (but
+   which must never appear in an enum when accessed by Rust code). The `Sentinel` tag is used by
+   the C++ wrapper classes to allow moving the ownership of an enum while invalidating the old copy.
+
+ * Struct member functions are mapped as `Struct_function_name` and take a reference to the mapped
+   struct as their first argument. Free-standing functions are mapped simply as `function_name` and
+   take the relevant mapped type arguments.
+
+   Functions may return a reference to an underlying Rust object with a mapped struct or an owned
+   Rust object with the same. The mapped struct contains a flag to indicate if the poitned-to Rust
+   object is owned or only a reference, and the object's corresponding free function will Do The
+   Right Thing based on the flag. In order to determine the expected return type, you should
+   reference the Rust documentation for the function.
+
+   Similarly, when a function takes an `Option<RustType>` as a parameter or a return value, the C
+   type is the same as if it took only `RustType`, with the `inner` field set to NULL to indicate
+   None. For example, `ChannelManager_create_channel` takes an `Option<LDKUserCOnfig>` not an
+   `LDKUserConfig`, but its definition is:
+   ```
+   MUST_USE_RES LDKCResult_NoneAPIErrorZ ChannelManager_create_channel(const LDKChannelManager *this_arg, ..., LDKUserConfig override_config);
+   ```
+
+ * Various containers (Tuples, Vecs, Results, etc) are mapped into C structs of the form
+   `LDKCContainerType_ContainerElementsZ`. Inner fields are often pointers, and in the case of
+   primitive types, these may be allocated in C using the system allocator. See [the Rust docs on
+   your platform's default System allocator](https://doc.rust-lang.org/std/alloc/struct.System.html)
+   for which allocator you must use.
+
+As the bindings are auto-generated, the best resource for documentation on them is the native Rust
+docs available via `cargo doc` or [docs.rs/lightning](https://docs.rs/lightning).
+
+The memory model is largely the Rust memory model and not a native C-like memory model. Thus,
+function parameters are largely only ever passed by reference or by move, with pass-by-copy
+semantics only applying to primitive types. However, because the underlying types are largely
+pointers, the same function signature may imply two different memory ownership semantics. Thus, you
+MUST read the Rust documentation while using the C bindings. For functions which ownership is moved
```

Comment:
"For functions which ownership is moved"
Not quite clear what that means. Functions that take arguments whose ownership is moved? Or closures whose ownership is moved?

Reply:
I tweaked it a bit, let me know what you think.